### ISSUE TITLE
`$.escape` is documented incorrectly

### ISSUE THREAD

**Issue by Foo-x** [oven-sh/bun#8752]
### What version of Bun is running?

1.0.26+c75e768a6

### What platform is your computer?

Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

Create a file named `foo.ts`.

```ts
import { $ } from 'bun'

const cmd = 'echo foo'
await $`${cmd}`
```

Then run `bun foo.ts`.

### What is the expected behavior?

To output `foo`.

### What do you see instead?

`bun: command not found: echo foo`

### Additional information

It doesn't work even if unescaping commands.

```ts
import { $ } from 'bun'

const cmd = 'echo foo'
await $`${$.escape(cmd)}`
```

`bun: command not found: "echo foo"`

**Comment by Foo-x** [comment]
I have found a workaround.

```ts
import { $ } from 'bun'

const cmd = ['echo', 'foo']
await $`${cmd}`
```

All I had to do was pass commands as an array, just like any other library API.
I think it would be better if it was clarified in the document.

**Comment by paperclover** [comment]
1. escape is documented wrong. it escapes the string. the correct way for this is `${{ raw: '...' }}`. i am changing this from a bug report to a documentation issue.

2. the array stuff i am not certain if it is correct behavior but this is discussed in #8550

**Comment by paperclover** [comment]
this section of the docs is 100% incorrect

https://bun.sh/docs/runtime/shell#escape-unescaped-strings

**Comment by zackradisic** [comment]
The docs are fixed as of: f77b217

**Comment by BalduinB** [comment]
I dont realy understand how to use the `escape` function.
lets say I have the following command I want to execute:

```ts 
const result = $`./some_script --arg1=${$.escape(input)} --arg2=${$.escape(arg2)}`
```

this doesn't work if the string contains eg. a  backtick (`)
am I missing something?
also tried
```ts 
const result = $`./some_script --arg1="${$.escape(input)}" --arg2="${$.escape(arg2)}"`
```

**Comment by zackradisic** [comment]
> I dont realy understand how to use the `escape` function. lets say I have the following command I want to execute:
> 
> ```ts
> const result = $`./some_script --arg1=${$.escape(input)} --arg2=${$.escape(arg2)}`
> ```
> 
> this doesn't work if the string contains eg. a backtick (`) am I missing something? also tried
> 
> ```ts
> const result = $`./some_script --arg1="${$.escape(input)}" --arg2="${$.escape(arg2)}"`
> ```

This looks like a bug with `$.escape()` with backticks, fix is here: #10980

Also note that substituting strings in a shell script will already escape it, so you should instead do

```ts
const result = $`./some_script --arg1=${input} --arg2=${arg2}`

```

**Comment by BalduinB** [comment]
thanks for the answer.
Looking forward to that fix!
any idea when it is getting merged? I kinda need it? or should I rather copy the escape function?
